DOCSP-39625: Add new bulk write API info (#12)
mongodb/docs-c: source/libmongoc/guides/bulk.txt

Executing write operations in batches reduces the number of network round trips, increasing write
throughput.

.. tip::

   Beginning in MongoDB Server version 8.0 and C Driver version 1.28, you can use the `mongoc_bulkwrite_t

>    <{+api-libmongoc+}/mongoc_bulkwrite_t.html>`__ type. This type uses the ``bulkWrite`` server command to

Author: Note: this API link will be broken until the 1.28 driver release

   { }

.. _bulkwrite_server:

``bulkWrite`` Server Command
----------------------------

The previous examples on this page use the `mongoc_bulk_operation_t <{+api-libmongoc+}/mongoc_bulk_operation_t.html>`__
type to perform bulk write operations. ``mongoc_bulk_operation_t`` runs the ``insert``, ``update`` and ``delete`` server

> commands. Bulk write operations that use the ``mongoc_bulk_operation_t`` type group operation types and sends them in

Reviewer: S: I think this can be cut down a bit to make it easier to understand
```suggestion
commands. Bulk write operations that use ``mongoc_bulk_operation_t`` group operations by type and sends them in
```

   { }

.. _bulkwrite_server:

``bulkWrite`` Server Command
----------------------------

The previous examples on this page use the `mongoc_bulk_operation_t <{+api-libmongoc+}/mongoc_bulk_operation_t.html>`__
type to perform bulk write operations. Alternatively, you can perform bulk writes by using the
`mongoc_bulkwrite_t <{+api-libmongoc+}/mongoc_bulkwrite_t.html>`__ type, which accesses the ``bulkWrite`` server
command.

`mongoc_bulkwrite_t <{+api-libmongoc+}/mongoc_bulkwrite_t.html>`__ uses the ``bulkWrite`` command
introduced in MongoDB Server 8.0. The ``bulkWrite`` command supports insert, update, and delete
operations in the same payload. You can also specify multiple collection namespaces in the same payload

> when using this command.

Author: I reworded this part similar to your suggestion, lmk what you think!

In contrast, `mongoc_bulk_operation_t <{+api-libmongoc+}/mongoc_bulk_operation_t.html>`__ uses the ``insert``,

> ``update`` and ``delete`` server commands available in all current MongoDB server versions. Write

Author: Yes, all versions including 8.0 - I got this copy from the C driver team but my understanding is: the `bulkWrite` command is only available in Server v8.0+, but the `insert`,
`update` and `delete` commands are available in all server versions. That includes version 8.0, but we should recommend the `bulkWrite` command instead if you're connected to 8.0

> ``update`` and ``delete`` server commands available in all current MongoDB server versions. Write

Author: I'll remove the mention of server versions
